pkg/api/types.go

```diff
@@ -1648,6 +1648,20 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+type Config struct {
+	TypeMeta   `json:",inline" yaml:",inline"`
+	ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Flags map[string]string `json:"flags,omitempty" yaml:"flags"`
```

Comment:
It is hard for me to evaluate this object without seeing an example of how it is used.  Can you come up with a sketch of a complete workflow involving a Config object and a Pod?  

I don't require anything as excellent and detailed as the https://github.com/GoogleCloudPlatform/kubernetes/blob/a78a0098770b88407fd48584558c674e218b4c34/docs/design/secrets.md by @pmorie but it is a good place to look for inspiration.